### PR TITLE
Added a configurable response RequiresHttpAction

### PR DESCRIPTION
**Pull request by garpinc**
Pass through error code to differential ajax error

### CODE REVIEW

File: pac4j-core/src/main/java/org/pac4j/core/exception/RequiresHttpAction.java [hunk 1]
```diff
@@ -108,6 +108,19 @@ public static RequiresHttpAction unauthorizedDigest(final String message, final
...
     
     /**
+     * Build a configurable response..
+     * 
+     * @param message message
+     * @param errorCode status code of response
+     * @param context context
+     * @return a configurable response
+     */
+    public static RequiresHttpAction configurable(final String message, int errorCode, final WebContext context) {
```
Comment (leleuj) [inline pac4j-core/src/main/java/org/pac4j/core/exception/RequiresHttpAction.java]: This already exists: https://github.com/pac4j/pac4j/blob/master/pac4j-core/src/main/java/org/pac4j/core/exception/RequiresHttpAction.java#L32

File: pac4j-http/src/main/java/org/pac4j/http/client/indirect/FormClient.java [hunk 2]
```diff
@@ -94,7 +94,7 @@ private RequiresHttpAction handleInvalidCredentials(final WebContext context, fi
 		// it's an AJAX request -> unauthorized (instead of a redirection)
 		if (getAjaxRequestResolver().isAjax(context)) {
 			logger.info("AJAX request detected -> returning " + errorCode);
-			return RequiresHttpAction.unauthorized("AJAX request -> " + errorCode, context, null);
+			return RequiresHttpAction.configurable("AJAX request -> " + errorCode, errorCode, context);
```
Comment (leleuj) [inline pac4j-http/src/main/java/org/pac4j/http/client/indirect/FormClient.java]: I think it's clear to have an explicit `unauthorized` here. And I think the `handleInvalidCredentials` method should no longer have an errorCode parameter.

Comment (garpinc) [inline pac4j-http/src/main/java/org/pac4j/http/client/indirect/FormClient.java]: you want do be able to distinguish between credentials being null and invalid credentials.. i.e 401 vs 403

Comment (leleuj) [inline pac4j-http/src/main/java/org/pac4j/http/client/indirect/FormClient.java]: OK

File: pac4j-core/src/main/java/org/pac4j/core/exception/RequiresHttpAction.java (hunk 1 above)
Comment (leleuj) [inline pac4j-core/src/main/java/org/pac4j/core/exception/RequiresHttpAction.java]: Please remove this one.

Comment (garpinc) [inline pac4j-core/src/main/java/org/pac4j/core/exception/RequiresHttpAction.java]: I did already

**Pull request merged**